redis.ts

```diff
@@ -1504,6 +1506,14 @@ export async function connect({
   const bufw = new BufWriter(conn);
   const exec = muxExecutor(bufr, bufw);
   const client = await create(conn, conn, conn, exec);
+  if (password) {
```

Comment:
Is empty string acceptable? I don't know redis support empty password for authentication, I think we should use `!= null` operator.

Reply:
Yes, it is the following reasons:

- In JavaScript, an empty string is evaluated as `false`.
- In Redis, setting `requirepass` to an empty string disables password authentication.

```shell
$ redis-cli
127.0.0.1:6379> config get requirepass
1) "requirepass"
2) ""
127.0.0.1:6379> config set requirepass foobar
OK
127.0.0.1:6379> quit


$ redis-cli
127.0.0.1:6379> exists "hoge"
(error) NOAUTH Authentication required.
127.0.0.1:6379> auth foobar
OK
127.0.0.1:6379> exists "hoge"
(integer) 0
127.0.0.1:6379> config set requirepass ""
OK
127.0.0.1:6379> quit


$ redis-cli
127.0.0.1:6379> exists "hoge"
(integer) 0
```

Reply:
In Redis, empty string is always denied.

So, Therefore, your point is correct :+1: